qiskit/assembler/assemble_circuits.py

```diff
@@ -105,18 +112,76 @@ def _assemble_circuit(circuit):
             del instruction._condition
 
         instructions.append(instruction)
-    return QasmQobjExperiment(instructions=instructions, header=header,
-                              config=config)
+    return (QasmQobjExperiment(instructions=instructions, header=header, config=config),
+            pulse_library)
+
+
+def _assemble_pulse_gates(circuit, run_config):
+    """Assemble and return the circuit calibrations and associated pulse library, if there are any.
+    The calibrations themselves may reference the pulse library which is returned as a dict.
+
+    Args:
+        circuit (QuantumCircuit): circuit which may have pulse calibrations
+        run_config (RunConfig): configuration of the runtime environment
+
+    Returns:
+        tuple(QasmExperimentCalibrations, dict(str, array)): The calibrations and pulse library.
```

Comment:
Perhaps we could type alias the pulse library?

Reply:
Can we write the return type as `Tuple[QasmExperimentCalibrations, Dict[str, List[complex]]]`?